src/server/api-server.js

```diff
@@ -0,0 +1,112 @@
+const Koa = require('koa');
+const Router = require('koa-router');
+const koaBody = require('koa-body');
+const uuid = require('uuid/v4');
+const Redux = require('redux');
+
+import { createGameReducer } from '../core/reducer';
+
+const createCredentials = () => uuid();
+const getGameMetadataKey = gameID => `${gameID}:metadata`;
+const getNewGameInstanceID = () => uuid();
+const createGameMetadata = () => ({
+  players: {},
+});
+
+export const isActionFromAuthenticPlayer = async ({
+  action,
+  db,
+  gameID,
+  playerID,
+}) => {
+  const gameMetadata = await db.get(getGameMetadataKey(gameID));
+  if (!gameMetadata) {
+    return true;
+  }
+
+  if (!action.payload) {
+    return true;
+  }
+
+  const hasCredentials = Object.keys(gameMetadata.players).some(key => {
+    return !!(
+      gameMetadata.players[key] && gameMetadata.players[key].credentials
+    );
+  });
+  if (!hasCredentials) {
+    return true;
+  }
+
+  if (!action.payload.credentials) {
+    return false;
+  }
+
+  if (
+    action.payload.credentials !== gameMetadata.players[playerID].credentials
+  ) {
+    return false;
+  }
+
+  return true;
+};
+
+export const createApiServer = ({ db, games }) => {
+  const app = new Koa();
+  const router = new Router();
+
+  router.post('/games/:id/create', koaBody(), async ctx => {
```

Comment:
This call should fail if the game already exists.

Reply:
@nicolodavis I don't think it's possible in this PR to create a game that already exists. 

A game has a name in its definition. The `create` endpoint takes that name, and creates a new instance of the game with that name, with a new game instance ID. 

There could be the unlikely event of a instance ID collision, but we could catch that internally and wouldn't need to fail the call.